### PR TITLE
Report failures if Windows build fails

### PR DESCRIPTION
**Pull request by jakule**
Currently, "Build on Windows" build does not report failures even if they happen. This PR fixes this behavior.

Force triggered build as "Build on Windows" is skipped on PR push:
https://github.com/gravitational/teleport/actions/runs/5390670986/jobs/9786370966

Depends on https://github.com/gravitational/teleport/pull/28357

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Review by webvictim — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Comment by jakule** [comment]
This PR is failing on the MQ (which means that my fix is working), but I need to wait for the Windows fix to hit master before I can merge that change.

**Pull request merged**
